src/_guides/language/language-tour.md

```diff
@@ -3730,56 +3780,29 @@ Note that the function's body doesn't need to use the Future API.
 Dart creates the Future object if necessary.
 
 
-<a id="await"></a>
-### Using await expressions with Futures
-
-An await expression has the following form:
-
-<pre>
-<b>await</b> <em>expression</em>
-</pre>
-
-You can use `await` multiple times in an async function.
-For example, the following code waits three times
-for the results of functions:
-
-<?code-excerpt "misc/lib/language_tour/async.dart (repeated-await)"?>
-{% prettify dart %}
-var entrypoint = await findEntrypoint();
-var exitCode = await runExecutable(entrypoint, args);
-await flushThenExit(exitCode);
-{% endprettify %}
-
-In <code>await <em>expression</em></code>,
-the value of <code><em>expression</em></code> is usually a Future;
-if it isn't, then the value is automatically wrapped in a Future.
-This Future object indicates a promise to return an object.
-The value of <code>await <em>expression</em></code> is that returned object.
-The await expression makes execution pause until that object is available.
-
-**If `await` doesn't work, make sure it's in an async function.**
-For example, to use `await` in your app's `main()` function,
-the body of `main()` must be marked as `async`:
+<a id="await-for"></a>
+### Handling Streams
 
-<?code-excerpt "misc/lib/language_tour/async.dart (main)"?>
-{% prettify dart %}
-Future main() async {
-  checkVersion();
-  print('In main: version is ${await lookUpVersion()}');
-}
-{% endprettify %}
+When you need to get values from a **Stream**,
+you have two options:
 
+* Use `async` and an _asynchronous for loop_ (`await for`).
+* Use the [Stream API](/guides/libraries/library-tour#stream).
 
-<a id="await-for"></a>
-### Using asynchronous for loops with Streams
+<aside class="alert alert-warning" markdown="1">
+**Note:**
+Asynchronous for loops aren't always appropriate.
+Use them only if it's OK for your program to do nothing
+while it waits for values from a single stream.
+</aside>
```

Comment:
I think futures are easier to comprehend than streams. When we were converting all of our examples from the Future API to async-await, we had no issues. It was harder to convert the examples that used Streams. I remember wanting to warn people about the stream issues, but I left out the warning (wish I could remember why).

Reply:
Actually, we did have the warning, but only in the library tour (https://www.dartlang.org/guides/libraries/library-tour#using-an-asynchronous-for-loop). It's arguably more appropriate here.

Reply:
I started cutting down the Future and Stream sections of the library tour—there's a lot of duplication, as you noted—but decided to leave that for a future PR.

Reply:
The new prose for the note is much better. I how about a slight simplification:

> Before using `await for`, ensure that it makes the code clearer
> and that you really do want to wait for _all_ of the stream's results
> (since some streams, like UI event streams. are potentially endless).